question24/main.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+const agreementLinkBtn = document.getElementById('js-agreement-link');
+const modal = document.getElementById('js-modal');
+const modalClose = document.getElementById('js-close');
+const mask = document.getElementById('js-mask');
+const checkbox = document.getElementById('js-checkbox');
+const submitButton = document.getElementById('js-submit-button');
+const target = document.querySelector(".last_content");
+
+
+agreementLinkBtn.addEventListener('click',()=>{
+    modal.classList.remove('hidden');
+    mask.classList.remove('hidden');
+});
+modalClose.addEventListener('click',()=>{
+    modal.classList.add('hidden');
+    mask.classList.add('hidden');
+});
+mask.addEventListener('click',()=>{
+    modalClose.click();
+});
+
+const options = {
+  root: document.getElementById('js-modal'),
+  threshold: 1
+};
+
+const checkWhenIntersect = (entries) => {
+  entries.forEach(entry => {
+    if (entry.isIntersecting) {
+      checkbox.checked = true;
+      checkbox.disabled = false;
+    }
+  });
+}
+
+const observer = new IntersectionObserver(checkWhenIntersect, options);
+observer.observe(target);
+
```

Comment:
IMO
It can be written this way.
It would be better if you add `js-`
```suggestion
const modalContainer = document.querySelector(".modal_container");
observer.observe(modalContainer.lastElementChild);
```

Reply:
Thank you for your review! Thanks for the nice idea! This is smarter than the before. Fixed so please confirm.
fixed commit : https://github.com/csakyo/homework/pull/38/commits/c775fb3dcbdd1f4854f5dd7e7d5208fa92f14fa0